src/shared/telemetry/awsTelemetryOptOut.ts

```diff
@@ -0,0 +1,127 @@
+            // if telemetry.enableTelemetry changed and user has not expressed a preference
+            if (event.affectsConfiguration('telemetry.enableTelemetry')
+                && this.settings.readSetting(AwsTelemetryOptOut.AWS_TELEMETRY_KEY) === undefined
+            ) {
+                await this.updateTelemetryConfiguration(TelemetryOptOutOptions.SameAsVsCode)
+            }
+        })
+    }
+
+    public async updateTelemetryConfiguration(response: TelemetryOptOutOptions) {
+        switch (response) {
+            case TelemetryOptOutOptions.Enable:
+                await this.settings.writeSetting(
+                    AwsTelemetryOptOut.AWS_TELEMETRY_KEY,
+                    true,
+                    vscode.ConfigurationTarget.Global
+                )
+                this.service.telemetryEnabled = true
+                break
+
+            case TelemetryOptOutOptions.Disable:
+                await this.settings.writeSetting(
+                    AwsTelemetryOptOut.AWS_TELEMETRY_KEY,
+                    false,
+                    vscode.ConfigurationTarget.Global
+                )
+                this.service.telemetryEnabled = false
+                break
+
+            case TelemetryOptOutOptions.SameAsVsCode:
+                await this.settings.writeSetting(
+                    AwsTelemetryOptOut.AWS_TELEMETRY_KEY,
+                    undefined,
+                    vscode.ConfigurationTarget.Global
+                )
+                const vsCodeTelemetryEnabled = this.getVSCodeTelemetrySetting()
+                this.service.telemetryEnabled = vsCodeTelemetryEnabled
+                break
+        }
+    }
+
+    /**
+     * Caution: you probably do not want to await this method
+     *
+     * This method awaits a showInfo call, which blocks until the user selects an option
+     * or explicitly cancels the dialog. 'Esc' on the dialog will continue to block, waiting for a response
+     * Ensure that you handle this suitably.
+     */
+    public async ensureUserNotified(): Promise<void> {
+        let response: string | undefined
+        if (!ext.context.globalState.get<boolean>(AwsTelemetryOptOut.TELEMETRY_OPT_OUT_SHOWN)) {
+            response = await this.showNotification()
+            await ext.context.globalState.update(AwsTelemetryOptOut.TELEMETRY_OPT_OUT_SHOWN, true)
+        } else {
+            response = this.settings.readSetting<string>(AwsTelemetryOptOut.AWS_TELEMETRY_KEY)
+        }
+        const enumValue = this.responseToOptionEnumValue(response)
+
+        await this.updateTelemetryConfiguration(enumValue)
+        this.service.notifyOptOutOptionMade()
+    }
+
+    public async showNotification(): Promise<string | undefined> {
+        const notificationMessage: string = localize(
+            'AWS.telemetry.notificationMessage',
+            'Please help improve the AWS Toolkit by enabling anonymous usage data to be sent to AWS. '
```

Comment:
Minor: Is this message the same in the other toolkits? The "Please" feels a little pushy to me.

Reply:
JetBrains: Anonymized usage metrics are collected by default. Click here to adjust this behavior.

where "here" takes you to the specific settings option

Reply:
Microsoft's prompt: Please help Microsoft improve Visual Studio Code by allowing the collection of usage data. Read our [privacy statement]({0}) for more details.

Reply:
Gotcha, sounds good.